internal/applicationsnapshot/report.go

```diff
@@ -289,6 +309,67 @@ func generateMarkdownSummary(r *Report) ([]byte, error) {
 	return markdownBuffer.Bytes(), nil
 }
 
+func generateTextReport(r *Report) ([]byte, error) {
+
+	mainTemplate := hd.Doc(`
+		Results
+		=======
+
+		{{ range .Components -}}
+		Name:       {{ .Name }}
+		ImageRef:   {{ .ContainerImage }}
+		Summary:    Violations: {{ len .Violations }}, Warnings: {{ len .Warnings }}, Successes: {{ .SuccessCount }}
+		{{ if gt (len .Violations) 0 -}}
+		Violations: {{ template "results" .Violations }}
+		{{ end }}
+		{{ if gt (len .Warnings) 0 -}}
+		Warnings: {{ template "results" .Warnings }}
+		{{ end }}
+		{{ if and (gt .SuccessCount 0) (.Successes) -}}
+		Successes: {{ template "results" .Successes }}
+		{{ end }}
+		{{ end }}{{/* range .Components */}}
+	`)
+
+	partials := map[string]string{
+		// For convenience since juggling the {{- and -}} gets tricky
+		"NL": "\n",
+
+		// Display a list of violations, warnings or successes
+		"results": `
+			{{ range . }}
+			* {{ red .Metadata.code }}
+			  {{ if ne .Message "Pass" }}{{ .Message }}
+			  {{ end -}}
+			  {{ .Metadata.title }}
+			  {{ .Metadata.description }}
+			  {{ if ne .Message "Pass" }}Solution: {{ .Metadata.solution }}
+			  {{ end -}}
+			{{ end }}
+		`,
+	}
+
+	t := template.New("t")
+
+	helpers := template.FuncMap{
+		"red":    utils.RedText,
+		"green":  utils.GreenText,
+		"yellow": utils.YellowText,
+	}
+
+	t.Funcs(helpers)
+
+	t = template.Must(t.Parse(mainTemplate))
+
+	for k, v := range partials {
+		t = template.Must(t.Parse(fmt.Sprintf("{{- define \"%s\" -}}\n%s\n{{- end -}}", k, hd.Doc(v))))
```

Comment:
Could put the nested templates in that same file that gets embedded, or use https://pkg.go.dev/text/template#ParseFS with embed to keep them in separate files